cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureVirtualMachineService.java

```diff
@@ -129,7 +145,7 @@ private AzureVirtualMachinesWithStatuses getVmsFromAzureAndFillStatusesIfResourc
         for (Map.Entry<String, Collection<String>> resourceGroupInstanceIdsMap : resourceGroupInstanceMultimap.asMap().entrySet()) {
             LOGGER.info("Get vms for resource group and add to all virtual machines: {}", resourceGroupInstanceIdsMap.getKey());
             try {
-                virtualMachines.putAll(getVirtualMachinesByName(azureClient,
+                virtualMachines.putAll(getVirtualMachinesByNameEmptyAllowed(azureClient,
```

Comment:
Just a note: since this is an intra-class invocation, the `@Retryable` on `getVirtualMachinesByNameEmptyAllowed()` will be ineffective. `getVmsAndVmStatusesFromAzure()` will surely provide retries (though with a different `maxAttempts` setting) , but e.g. `getVmsAndVmStatusesFromAzureWithoutRetry()` will not exhibit any retries at all.

Reply:
Yes that correct and in this context this might be misleading, but it's called by the context of the syncer where we don't want any retries.
Might be better to make this one private and remove the annotation.